mathesar_ui/src/systems/table-view/table-inspector/column/ColumnMode.svelte

```diff
@@ -12,58 +11,63 @@
   const tabularData = getTabularDataStoreFromContext();
   $: ({ processedColumns, selection } = $tabularData);
   $: ({ selectedCells } = selection);
-  $: selectedColumnsId = $selectedCells
-    .valuesArray()
-    .map((cell) => getSelectedColumnId(cell));
-  $: uniquelySelectedColumnsId = Array.from(new Set(selectedColumnsId));
-
-  function getSelectedColumn(selectedColumnId: number): ProcessedColumn {
-    return $processedColumns.get(selectedColumnId)!;
-  }
+  $: selectedColumns = (() => {
+    const ids = new Set([...$selectedCells].map(getSelectedColumnId));
+    const columns = [];
+    for (const id of ids) {
+      const c = $processedColumns.get(id);
+      if (c !== undefined) {
+        columns.push(c);
+      }
+    }
+    return columns;
+  })();
```

Comment:
Nice use of self-executing functions for reactivity 👌

I wonder why we haven't done this before in many places in our codebase.